pandas/core/generic.py

```diff
@@ -1084,6 +1084,10 @@ def to_hdf(self, path_or_buf, key, **kwargs):
                        / selecting subsets of the data
         append   : boolean, default False
             For Table formats, append the input data to the existing
+        data_columns : list of columns, or True, default None
+            This will create additional indexed columns for on-disk queries,
+            by default only 'index' and 'columns' are indexed. True will index
+            all columns.
```

Comment:
How can we make this specific to Series,DataFrame and Panel? Because considering the complexity of pandas I find it highly important to have the docstrings as useful as possible.